skyportal/models.py

```diff
@@ -2851,6 +2895,51 @@ class CommentOnSpectrum(Base, CommentMixin):
 )
 
 
+class AnnotationOnSpectrum(Base, AnnotationMixin):
+
+    __tablename__ = 'annotations_on_spectra'
+
+    create = AccessibleIfRelatedRowsAreAccessible(obj='read', spectrum='read')
+
+    read = accessible_by_groups_members & AccessibleIfRelatedRowsAreAccessible(
+        obj='read',
+        spectrum='read',
+    )
+
+    update = delete = AccessibleIfUserMatches('author')
+
+    spectrum_id = sa.Column(
+        sa.ForeignKey('spectra.id', ondelete='CASCADE'),
+        nullable=False,
+        index=True,
+        doc="ID of the Annotation's Spectrum.",
+    )
+    spectrum = relationship(
+        'Spectrum',
+        back_populates='annotations',
+        doc="The Spectrum referred to by this annotation.",
+    )
+
+    __table_args__ = (UniqueConstraint('spectrum_id', 'origin'),)
```

Comment:
```suggestion
    __table_args__ = (UniqueConstraint('spectrum_id', 'origin'))
```

Reply:
Without the comma it complains it needs a tuple. I'm leaving it in.